src/main/java/com/metamx/emitter/core/ParametrizedUriEmitter.java

```diff
@@ -38,8 +37,8 @@ private static UriExtractor makeUriExtractor(ParametrizedUriEmitterConfig config
    */
   private final ConcurrentHashMap<URI, HttpPostEmitter> emitters = new ConcurrentHashMap<>();
   private final UriExtractor uriExtractor;
-  private final AtomicBoolean startFlag = new AtomicBoolean(false);
-  private final Object closeLock = new Object();
+  private final Object startCloseLock = new Object();
+  private boolean started = false;
```

Comment:
Suggest adding explicit documentation saying this is guarded by the startCloseLock, and anything that would ever need to check the stared or closed states needs to acquire the lock first.